src/index.js

```diff
@@ -59,17 +59,47 @@ class FloodSub extends EventEmitter {
     this._dialPeer = this._dialPeer.bind(this)
   }
 
+  _addPeer (peer) {
+    const id = peer.info.id.toB58String()
+
+    // Always use an existing peer.
+    let existing = this.peers.get(id)
+    if (existing !== undefined) {
+      log('already existing peer', id)
+      ++existing._references
+    } else {
+      log('new peer', id)
+      this.peers.set(id, peer)
+      existing = peer
+    }
```

Comment:
This if can be inverted for great readability (by this I mean, avoid do negative equals with things like undefined or null)

```
let existing = this.peers.get(id)
if (!existing) {
  log('new peer', id)
  this.peers.set(id, peer)
  existing = peer
} else {
  log('already existing peer', id)
  ++existing._references
}
```

Reply:
Good point, positive logic is always better.

Reply:
even better
```
    if (existing) {
      log('already existing peer', id)
      ++existing._references
    } else {
      log('new peer', id)
      this.peers.set(id, peer)
      existing = peer
    }```